csharp/src/Google.Protobuf/JsonTokenizer.cs

```diff
@@ -471,9 +471,18 @@ private double ReadNumber(char initialCharacter)
                 // TODO: What exception should we throw if the value can't be represented as a double?
                 try
                 {
-                    return double.Parse(builder.ToString(),
+                    double result = double.Parse(builder.ToString(),
                         NumberStyles.AllowLeadingSign | NumberStyles.AllowDecimalPoint | NumberStyles.AllowExponent,
                         CultureInfo.InvariantCulture);
+
+                    // .NET Core 3.0 and later returns infinity if the number is too large or small to be represented.
+                    // For compatibility with other Protobuf implementations the tokenizer should still throw.
+                    if (double.IsInfinity(result))
```

Comment:
FTR I wasn't sure if this doesn't break parsing of the "Infinity" and "-Infinity" values, but it seems that it does not as  verified by this test: https://github.com/protocolbuffers/protobuf/blob/47e05427e3341c9b18fff047e7d9f79af0dafe9b/csharp/src/Google.Protobuf.Test/JsonParserTest.cs#L356